android/src/main/java/com/swmansion/rnscreens/ScreenStack.java

```diff
@@ -171,8 +171,20 @@ protected void performUpdate() {
       }
     }
 
-    for (ScreenStackFragment screen : mScreenFragments) {
-      // detach all screens that should not be visible
+    // detach all screens that should not be visible
+    for (int i = 0; i < mScreenFragments.size(); i++) {
+      ScreenStackFragment screen = mScreenFragments.get(i);
+
+      // Stop detaching screens if next screen is a transparent modal. This is needed to be able to
+      // see all screens above and including the last non transparent screen
+      boolean hasNextScreen = i + 1 < mScreenFragments.size();
+      if(hasNextScreen){
+        ScreenStackFragment nextScreen = mScreenFragments.get(i + 1);
+        if(nextScreen.getScreen().getStackPresentation() == Screen.StackPresentation.TRANSPARENT_MODAL){
```

Comment:
this seem not to handle situation when you go back from non-transparent to transparent modal. To support the case when you can see through we used the `belowTop` variable. Apparently it only supports the case when you have a single level of transparent layers. In order to support multiple layers (what I understand this PR is addressing) it should modify all the logic that deals with `belowTop` including fragments where we attach `belowTop` (when we should attach possible multiple views) and bring them to front in the right order to maintain the proper z-order when attached.